roles/openldap/tasks/install-configure.yml

```diff
@@ -0,0 +1,80 @@
+- name: install the openldap packages
+  apt: name={{ item }} state=installed
+  with_items: "{{ openldap.ldap_pkgs }}"
+
+- name: create directory for ldif files
+  file: path={{ openldap.config_dir }}
+        state=directory
+        owner=root
+        group=root
+        mode=0400
+
+- name: copy templates
+  template: src={{ item }}
+            dest={{ openldap.config_dir }}/{{ item | basename }}
+  with_fileglob: ../templates/install_configure/*
+
+- name: set log level
+  shell: ldapmodify -Q -Y EXTERNAL -H ldapi:/// -f {{ openldap.config_dir }}/logging.ldif
+
+- name: get admin/suffix dn
+  shell: ldapsearch -Q -LLL -Y EXTERNAL -H ldapi:/// -s sub -b "{{ openldap.suffix }}" cn=admin
+  register: admin_name
+  ignore_errors: yes
+  changed_when: False
+
+- name: ensure admin/suffix is created
+  shell: ldapmodify -Y EXTERNAL -H ldapi:/// -f {{ openldap.config_dir }}/set_admin.ldif
+  when: admin_name.stdout.count("cn={{ openldap.root_cn }}") < 1
+
+- name: get base domain
+  shell: ldapsearch -D {{ openldap.root_dn }} -w {{ openldap.root_password }}  -s sub -b "{{ openldap.suffix }}" objectclass=*
+  register: base_domain
+  ignore_errors: yes
+  changed_when: False
+
+- name: ensure base domain is created
+  shell: ldapadd -D {{ openldap.root_dn }} -w {{ openldap.root_password }} -f {{ openldap.config_dir }}/base_domain.ldif
```

Comment:
Can be done, but then we are deciding if something is an error or not based on the error string, a valid error may have the string we are looking for, the possible error outcomes and their corresponding messages are not completely known.
We can probably also go by return codes of the shell commands, but this will need more extensive testing.